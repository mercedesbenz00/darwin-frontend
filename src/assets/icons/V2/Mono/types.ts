export enum IconVariant {
  DEFAULT = 'default',
  DISABLED = 'disabled',
  SELECTED = 'selected',
  NEGATIVE = 'negative',
}
