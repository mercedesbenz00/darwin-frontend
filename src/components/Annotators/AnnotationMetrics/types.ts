import { MembershipPayload, DatasetAnnotationReportPayload } from '@/store/types'

export type AnnotationDataRange = 'total' | 'month' | 'week' | 'day'
export type AnnotationDataGranularity = 'day' | 'hour'

export type ChartDimension =
  'annotationTime' |
  'imagesAnnotated' |
  'avgTimePerAnnotation' |
  'annotationsCreated' |
  'reviewPassRate'

export type AdaptedDataPoint =
  {[k in Exclude<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number } &
  {[k in Extract<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number | null} &
  { userId: number | null, date: string }

export type AggregatedDataPoint = Pick<
    DatasetAnnotationReportPayload,
    | 'annotation_time'
    | 'annotations_approved'
    | 'annotations_created'
    | 'date'
    | 'images_annotated'
    | 'images_approved'
    | 'images_rejected'
    | 'user_id'
  >

export type AnnotationDimensionData = {
  annotationTime: number,
  annotationsCreated: number,
  avgTimePerAnnotation: number | null,
  imagesAnnotated: number,
  reviewPassRate: number | null
}

export type AnnotationChartData = {
  data: AdaptedDataPoint[]
  dimensions: AnnotationDimensionData
  member?: MembershipPayload
  visible: boolean
}

export type AnnotationBarChartData = {
  data: AggregatedDataPoint[]
  member?: MembershipPayload
}

/**
 * Format for a single item to be shown in a chart tooltip.
 *
 * A tooltip will contain multiple items
 */
export type ChartTooltipData = { member: MembershipPayload | null, value: string}
