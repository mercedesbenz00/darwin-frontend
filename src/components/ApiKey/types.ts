export type AbilityGroup =
  'manage_comments' |
  'manage_tasks' |
  'manage_datasets' |
  'manage_team' |
  'manage_team_members' |
  'manage_models'

export type Group = { id: AbilityGroup; name: string }
export type GroupOption = Group & { selected: boolean }

export enum SupportedLanguages {
  shell = 'shell',
  python = 'python',
  javascript = 'javascript',
  elixir = 'elixir',
  cli = 'cli'
}

export type AccessCodeExample = { language: SupportedLanguages, code: string }
