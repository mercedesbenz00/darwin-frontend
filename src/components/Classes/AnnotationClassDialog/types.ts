import { AnnotationClassPayload, AnnotationClassMetadata, AnnotationTypeName } from '@/store/types'

type PendingFields =
  | 'crop_key'
  | 'crop_url'
  | 'image_height'
  | 'image_width'
  | 'key'
  | 'original_image_url'
  | 'scale'
  | 'x'
  | 'y'

/**
 * Holds a class image that isn't necessarily persisted to the backend yet.
 * That means it might be missing any of the url fields
 *
 * It can be saved to backend once at least the `key` field is set, as that one
 * can be used to infer/compute all the other fields.
 *
 * - `crop_key` - repleace "original" with "crop" in `key` field
 * - `crop_url` - signed s3 url (by backend), for `crop_key`
 * - `original_image_url` - signed s3 url (by backend) for `key`
 */
export type PendingClassImage =
  Omit<AnnotationClassPayload['images'][0], PendingFields>
  & Partial<Pick<AnnotationClassPayload['images'][0], PendingFields>>

// eslint-disable-next-line camelcase
export type PendingClassImageWithUrlSet = PendingClassImage & { original_image_url: string }

export const isClassImage = (
  image: PendingClassImage
): image is AnnotationClassPayload['images'][0] =>
  !!image.key && !!image.crop_key

export type AnnotationClassData = {
  datasets: AnnotationClassPayload['datasets']
  description: string | null
  hotkey: null | string
  id: number | null
  images: PendingClassImage[]
  metadata: AnnotationClassMetadata
  name: string | null
}

export type DefaultAnnotationClassData = Partial<AnnotationClassData> & {
  selectedMainAnnotationType?: AnnotationTypeName
}

export type AnnotationClassValidationErrors = {
  annotationTypes?: string
  name?: string
  skeleton?: string
}

export const VIEWPORT_WIDTH: number = 150
export const VIEWPORT_HEIGHT: number = 150

export type CropInfo = {
  orientation: number
  points: [string, string, string, string]
  zoom: number
}

/**
 * VueCroppie is really archaic and requires us to get crop info via $ref,
 * so this type defines that ref
 */
export type Croppie = {
  /** Binds options to the underlying croppie library */
  bind: (options: Pick<CropInfo, 'points' | 'zoom'> & { url: string}) => void

  /** Retrieves current crop info */
  get: () => CropInfo
  result: (options?: Record<string, string>, callback?: Function) => void

  /** Sets scale/zoom level of cropper */
  setZoom: (value: number) => void
}
