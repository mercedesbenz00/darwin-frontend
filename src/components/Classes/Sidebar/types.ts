import { AnnotationTypeName } from '@/store/types'

export type AnnotationTypeFilterItemType = {
  id: string | number
  label: string
  icon?: string
  name: AnnotationTypeName
  color: string
  count: number | null
  [key: string]: any
}
