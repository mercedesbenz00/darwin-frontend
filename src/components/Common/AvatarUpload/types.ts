export type AvatarUploadData = {
  hash: string
  file: File
  type: string | null
}
