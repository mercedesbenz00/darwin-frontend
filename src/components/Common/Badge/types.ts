import { RGBA } from '@/utils'

export type ParsedColorType = {
  bg: string,
  text: string
}

export type BadgeType = {
  id?: string,
  label: string,
  color?: RGBA
  size?: 'small' | 'medium' | 'large'
  highContrast?: boolean
}

export enum BadgeSize {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
}

export enum BadgeTag {
  A = 'a',
  DIV = 'div',
  ROUTER_LINK = 'router-link',
  BUTTON = 'button'
}

export interface BadgesProps {
  items: BadgeType[],
  selected?: boolean,
  wrap?: boolean
}

export interface BadgeProps {
  label: string,
  color?: RGBA,
  size?: string,
  tag?: string,
  tooltip?: string,
  highContrast?: boolean,
  overrideLabelColor?: string,
  deletable?: boolean,
  noTooltip?: boolean,
  alpha?: number
}
