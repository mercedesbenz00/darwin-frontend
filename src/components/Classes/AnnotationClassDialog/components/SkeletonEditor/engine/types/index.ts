export * from './SkeletonEdgeType'
export * from './SkeletonNodeType'
