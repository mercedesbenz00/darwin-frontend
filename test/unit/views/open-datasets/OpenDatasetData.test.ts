import { createLocalVue, shallowMount, Stubs } from '@vue/test-utils'
import Vuex from 'vuex'

import createTestStore from 'test/unit/createTestStore'
import {
  buildDatasetPayload,
  buildDatasetDetailPayload,
  buildDatasetItemPayload
} from 'test/unit/factories'
import { DatasetDetailLayout } from 'test/unit/stubs'

import { DatasetItemStatus } from '@/store/types'
import OpenDatasetData from '@/views/open-datasets/OpenDatasetData.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

const myDataset = buildDatasetPayload({
  id: 1,
  name: 'My Dataset',
  slug: 'my-dataset',
  public: false,
  active: true,
  archived_at: null,
  archived: false,
  instructions: '',
  progress: 0.5,
  owner_id: 1,
  work_size: 20,
  thumbnails: [],
  num_classes: 3,
  num_annotations: 1000,
  num_images: 4,
  team_id: 1
})

const myDatasetDetails = buildDatasetDetailPayload({ id: 1 })

const items = [
  buildDatasetItemPayload({ id: 1, seq: 1, status: DatasetItemStatus.annotate }),
  buildDatasetItemPayload({ id: 2, seq: 2, status: DatasetItemStatus.uploading }),
  buildDatasetItemPayload({ id: 3, seq: 3, status: DatasetItemStatus.complete }),
  buildDatasetItemPayload({ id: 4, seq: 4, status: DatasetItemStatus.archived })
]

const stubs: Stubs = { OpenDatasetDetailLayout: DatasetDetailLayout, 'router-view': true }

let store: ReturnType<typeof createTestStore>

beforeEach(() => { store = createTestStore() })

it('matches the snapshot in dataset route', () => {
  store.commit('dataset/SET_CURRENT_DATASET_DETAILS', myDatasetDetails)
  store.commit('dataset/SET_DATASET_ITEMS', items)
  store.commit('dataset/SET_CURRENT_DATASET_LOADED')
  store.commit('dataset/SET_DATASET_ITEMS_LOADED')

  const mocks = { $route: { params: {} } }
  const propsData = { dataset: myDataset }
  const wrapper = shallowMount(OpenDatasetData, { localVue, mocks, propsData, store, stubs })

  expect(wrapper).toMatchSnapshot()
})

it('matches the snapshot in dataset folder route', () => {
  store.commit('dataset/SET_CURRENT_DATASET_DETAILS', myDatasetDetails)
  store.commit('dataset/SET_DATASET_ITEMS', items)
  store.commit('dataset/SET_CURRENT_DATASET_LOADED')
  store.commit('dataset/SET_DATASET_ITEMS_LOADED')

  const mocks = { $route: { params: { path: '/root' } } }
  const propsData = { dataset: myDataset }
  const wrapper = shallowMount(OpenDatasetData, { localVue, mocks, propsData, store, stubs })

  expect(wrapper).toMatchSnapshot()
})
