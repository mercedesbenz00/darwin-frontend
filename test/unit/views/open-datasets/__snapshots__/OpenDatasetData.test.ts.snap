// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot in dataset folder route 1`] = `
<div class="dataset-details" dataset="[object Object]">
  <dataset-items-loader-stub dataset="[object Object]"></dataset-items-loader-stub>
  <dataset-channel-subscriber-stub dataset="[object Object]"></dataset-channel-subscriber-stub>
  <dataset-item-gallery-stub dataset="[object Object]" readonly="true"></dataset-item-gallery-stub>
  <read-only-context-menu-stub></read-only-context-menu-stub>
  <open-data-sidebar-stub dataset="[object Object]"></open-data-sidebar-stub>
</div>
`;

exports[`matches the snapshot in dataset route 1`] = `
<div class="dataset-details" dataset="[object Object]">
  <dataset-items-loader-stub dataset="[object Object]"></dataset-items-loader-stub>
  <dataset-channel-subscriber-stub dataset="[object Object]"></dataset-channel-subscriber-stub>
  <dataset-item-gallery-stub dataset="[object Object]" readonly="true"></dataset-item-gallery-stub>
  <read-only-context-menu-stub></read-only-context-menu-stub>
  <open-data-sidebar-stub dataset="[object Object]"></open-data-sidebar-stub>
</div>
`;
